### PR TITLE
Allow user to customize query cost estimate

### PR DESCRIPTION
**Pull request by betodealmeida**
### CATEGORY

Choose one

- [ ] Bug Fix
- [X] Enhancement (new features, refinement)
- [ ] Refactor
- [ ] Add tests
- [ ] Build / Development Environment
- [ ] Documentation

### SUMMARY

https://github.com/apache/incubator-superset/pull/8172 introduced the ability to estimate query costs before running them. Currently, this is supported by Presto, and it returns a _relative_ CPU and network cost.

Ideally, we want to show the cost as dollars, time or percentile, instead of showing relative costs. Since the relationship between the relative cost and the actual cost varies from infrastructure to infrastructure, we need to provide users a way of customizing the query cost estimate.

Here's a hypothetical example where we use a coefficient to compute the query cost in dollars based on CPU and network cost:

```python
def presto_query_cost_formatter(result):
    cpu_coefficient = 1e-12
    network_coefficient = 1e-12

    cost = 0
    for row in result:
        cost += row.get("cpuCost", 0) * cpu_coefficient
        cost += row.get("networkCost", 0) * network_coefficient

    return [{"Cost": f"US$ {cost:.2f}"}]


DEFAULT_FEATURE_FLAGS = {
    "ESTIMATE_QUERY_COST": True,
    "QUERY_COST_FORMATTER": {"presto": presto_query_cost_formatter},
}
```

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

Screen Shot 2019-10-29 at 1 36 24 PM

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### REVIEWERS

@khtruong

### CODE REVIEW

File: docs/sqllab.rst [hunk 1]
```diff
@@ -123,6 +123,29 @@ database configuration:
 Here, "version" should be the version of your Presto cluster. Support for this
 functionality was introduced in Presto 0.319.
 
+You also need to enable the feature flag in your `superset_config.py`, and you
+can optionally specify a custom formatter. Eg:
+
+.. code-block:: python
+
+	def presto_query_cost_formatter(result):
```
Comment (etr2460) [inline docs/sqllab.rst]: could we add a type to this so that people know what valid params there are? Or add that in a comment somewhere?

**Review by etr2460 — COMMENTED** [review]
one documentation request!

File: docs/sqllab.rst (hunk 1 above)
Comment (betodealmeida) [reply docs/sqllab.rst]: Done! Sorry, I'm still not fully used to types in Python, and I keep forgetting them. :-P

**Review by etr2460 — APPROVED** [review]
lgtm, thanks for the docs!

**Pull request merged**